Admin/Get-MigrationReport.ps1

```diff
@@ -0,0 +1,236 @@
+    } elseif ($failureMessage.contains( "MapiExceptionTooComplex: Unable to query table rows")) {
+        Write-Host -ForegroundColor Red "Error: MapiExceptionTooComplex: Unable to query table rows. Reference: https://aka.ms/AAjey20 "
+        Add-Content -Path $LogFile -Value "Error: MapiExceptionTooComplex: Unable to query table rows. Reference: https://aka.ms/AAjey20 "
+    } elseif ($failureMessage.Contains("Mailbox Replication Proxy Service can't process this request because it has reached the maximum number of active MRS connections allowed")) {
+        Write-Host -ForegroundColor Red "Error: Mailbox Replication Proxy Service can't process this request because it has reached the maximum number of active MRS connections allowed. Reference: https://aka.ms/AAjdrda "
+        Add-Content -Path $LogFile -Value "Error: Mailbox Replication Proxy Service can't process this request because it has reached the maximum number of active MRS connections allowed. Reference: https://aka.ms/AAjdrda "
+    }
+}
+
+
+#===================MAIN======================
+New-Item $OutputFolder -ItemType Directory -Force | Out-Null
+New-Item $LogFile -Type File -Force -ErrorAction SilentlyContinue | Out-Null
+
+foreach ($Mailbox in $Identity) {
+    $MoveRequest = Get-MoveRequest $Mailbox -ErrorAction SilentlyContinue
+    $MoveRequestStatistics = Get-MoveRequestStatistics $Mailbox -IncludeReport -ErrorAction SilentlyContinue
+    if ($null -eq $MoveRequest) {
+        Write-Host -ForegroundColor Red -Value "[ERROR] The MoveRequest for the $Mailbox cannot be found, please check spelling and try again!"
+        Add-Content -Path $LogFile -Value "[ERROR] The MoveRequest for the $Mailbox cannot be found, please check spelling and try again!"
+    }
+    $Batch = $MoveRequestStatistics.BatchName
+    $MigrationBatch = Get-MigrationBatch $Batch -IncludeReport  -ErrorAction SilentlyContinue
+    $UserMigration = Get-MigrationUser $Mailbox  -ErrorAction SilentlyContinue
+    $UserMigrationStatistics = Get-MigrationUserStatistics $Mailbox -IncludeSkippedItems -IncludeReport -ErrorAction SilentlyContinue
+    $Endpoint = $MigrationBatch.SourceEndpoint
+    $MigrationEndPoint = Get-MigrationEndpoint -Identity $Endpoint -DiagnosticInfo Verbose -ErrorAction SilentlyContinue
+    $MailboxStatistics = Get-MailboxStatistics $Mailbox -IncludeMoveReport -IncludeMoveHistory -ErrorAction SilentlyContinue
+    $UniqueFailure = $MoveRequestStatistics.Report.Failures | Select-Object FailureType -Unique
+    $DetailedFailure = foreach ($U in $UniqueFailure) { $MoveRequestStatistics.Report.Failures | Where-Object { $_.FailureType -like $U.FailureType } | Select-Object Timestamp, FailureType, FailureSide, Message -Last 1 | Format-Table -Wrap }
+    $FailureMessages = @(); foreach ($U in $UniqueFailure) { $FailureMessages += $MoveRequestStatistics.Report.Failures | Where-Object { $_.FailureType -like $U.FailureType } | select -Last 1 }; $message = $FailureMessages[0].Message
```

Comment:
This can be cleaned up, so you only loop through the list once vs twice.

Reply:
Actually, I tried using one variable, but I couldn't get the if statement to work. the second line to convert the array to a string so that it may subsequently be compared. please let me know if you have better ideas here!